src/test/java/com/palantir/docker/compose/connection/waiting/SuccessOrFailureShould.java

```diff
@@ -50,4 +50,30 @@ public void fail_from_an_exception() {
                 containsString("oh no")
             ))));
     }
+
+    @Test
+    public void succeed_on_a_lambda_that_returns_true() {
+        SuccessOrFailure successFromLambda = SuccessOrFailure.onResultOf(() -> true);
+        assertThat(successFromLambda, is(successful()));
+    }
+
+    @Test
+    public void fail_on_a_lambda_that_throws_an_exception() {
+        SuccessOrFailure failureFromLambda = SuccessOrFailure.onResultOf(() -> {
+            throw new IllegalArgumentException("oh no");
+        });
+
+        assertThat(failureFromLambda,
+                is(failureWithMessage(both(
```

Comment:
This stuff is always just going to generate error messages, it's the only reason for it existing.